### PR TITLE
[New Rule] System Time Discovery

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Detects the usage of commonly used system time discovery techniques, which attackers may use during the reconnaissance phase after compromising a system.

### CODE REVIEW

File: rules/windows/discovery_system_time_discovery.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+[metadata]
+creation_date = "2023/01/17"
+integration = ["windows"]
```
Comment (Samirbous) [inline rules/windows/discovery_system_time_discovery.toml:3]:
```suggestion
integration = ["windows", "endpoint"]
```

rules based on process creation events are compatible with `endpoint` integration too (Endpoint events)

File: rules/windows/discovery_system_time_discovery.toml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+min_stack_version = "8.3.0"
+updated_date = "2023/01/17"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects the usage of commonly used system time discovery techniques, which attackers may use during the reconnaissance phase after compromising a system. 
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
```
Comment (Samirbous) [inline rules/windows/discovery_system_time_discovery.toml:15]:
```suggestion
index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"]
```

**Review by Samirbous — CHANGES_REQUESTED** [review]
Nice first rule, letf some suggestions

File: rules/windows/discovery_system_time_discovery.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/discovery_system_time_discovery.toml:15]: @Aegrah, I forgot these. We need `logs-windows.*` for sysmon compatibility and endgame for endgame EDR compatibility

File: rules/windows/discovery_system_time_discovery.toml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#### Possible investigation steps
+- Check whether the user running these commands often accesses the host, and should have access to this system.
+- Investigate whether possible malicious behavior is spotted before and after the timestamp correlated with this alert.
+- Contact the user to ensure that the executed command was intentional benign behavior.
+### False positive analysis
+- Regular users should rarely look up their current timezones, and even more rarely do this through command line utilities. Administrators might leverage command line utilities to do so. Investigate whether the user running these commands did this intentionally. In the case of authorized benign true positives (B-TPs), exceptions can be added. 
+"""
```
Comment (w0rk3r) [inline rules/windows/discovery_system_time_discovery.toml:28]:
```suggestion
```
As we talked, let's keep this one without a guide for now

File: rules/windows/discovery_system_time_discovery.toml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1124"
+name = "System Time Discovery"
+reference = "https://attack.mitre.org/techniques/T1124/"
+
+[rule.threat.tactic]
+id = "TA0007"
+name = "Discovery"
+reference = "https://attack.mitre.org/tactics/TA0007/"
```
Comment (w0rk3r) [inline rules/windows/discovery_system_time_discovery.toml:54]:
```suggestion
reference = "https://attack.mitre.org/tactics/TA0007/"

```

File: rules/windows/discovery_system_time_discovery.toml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Discovery", "Elastic Endgame"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and
+(
+ ((process.name: "net.exe" or (process.name : "net1.exe" and not process.parent.name : "net.exe")) and process.args : "time") or process.name: "w32tm.exe" and process.args: "/tz" or 
+ process.name: "tzutil.exe" and process.args: "/g"
+)
```
Comment (Samirbous) [inline rules/windows/discovery_system_time_discovery.toml:31]:
```suggestion
process where event.type == "start" and
(
 ((process.name: "net.exe" or (process.name : "net1.exe" and not process.parent.name : "net.exe")) and process.args : "time") or 
 (process.name: "w32tm.exe" and process.args: "/tz") or 
 (process.name: "tzutil.exe" and process.args: "/g")
)
```

Comment (Samirbous) [reply rules/windows/discovery_system_time_discovery.toml:31]: `process where (A or B or (C and D))` C and D should be between `( and )`

**Review by Samirbous — APPROVED** [review]

**Review by w0rk3r — APPROVED** [review]
Nice work

**Pull request merged**